code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_ai.dm

```diff
@@ -0,0 +1,310 @@
+			else if(ai_ventcrawls && world.time > (last_ventcrawl_time + my_ventcrawl_freq))
+				if(prob(idle_ventcrawl_chance))
+					last_ventcrawl_time = world.time
+					var/vdistance = 99
+					for(var/obj/machinery/atmospherics/unary/vent_pump/v in view(10, src))
+						if(!v.welded)
+							if(get_dist(src,v) < vdistance)
+								entry_vent = v
+								vdistance = get_dist(src,v)
+					if(entry_vent)
+						path_to_vent = 1
+			else
+				// If none of the general actions apply, check for class-specific actions.
+				spider_special_action()
+		else if(AIStatus != AI_OFF && target)
+			CreatePath(target)
+	..()
+
+/mob/living/simple_animal/hostile/poison/terror_spider/adjustBruteLoss(damage)
+	..(damage)
+	Retaliate()
+
+/mob/living/simple_animal/hostile/poison/terror_spider/adjustFireLoss(damage)
+	..(damage)
+	Retaliate()
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/Retaliate()
+	var/list/around = oview(src, 7)
+	var/list/ts_nearby = list()
+	for(var/atom/movable/A in around)
+		if(A in enemies)
+			continue
+		if(isterrorspider(A))
+			ts_nearby += A
+			continue
+		if(isliving(A))
+			var/mob/living/M = A
+			if(!("terrorspiders" in M.faction))
+				enemies |= M
+				visible_message("<span class='danger'>[src] glares at [M]! </span>")
+		else if(istype(A, /obj/mecha))
+			var/obj/mecha/M = A
+			if(M.occupant)
+				enemies |= M
+				enemies |= M.occupant
+		else if(istype(A, /obj/spacepod))
+			var/obj/spacepod/M = A
+			if(M.pilot)
+				enemies |= M
+				enemies |= M.pilot
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/H in ts_nearby)
+		var/retaliate_faction_check = 0
+		for(var/F in faction)
+			if(F in H.faction)
+				retaliate_faction_check = 1
+				break
+		if(retaliate_faction_check && !attack_same && !H.attack_same)
+			H.enemies |= enemies
+	return 0
+
+// --------------------------------------------------------------------------------
+// --------------------- TERROR SPIDERS: PATHING CODE -----------------------------
+// --------------------------------------------------------------------------------
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/CreatePath(mygoal)
+	var/m2 = get_turf(src)
+	if(m2 == mylocation)
+		chasecycles++
+		ClearObstacle(get_turf(mygoal))
+		if(chasecycles >= 2)
+			chasecycles = 0
+			if(spider_opens_doors)
+				var/tgt_dir = get_dir(src,mygoal)
+				for(var/obj/machinery/door/airlock/A in view(1, src))
+					if(A.density)
+						try_open_airlock(A)
+				for(var/obj/machinery/door/firedoor/F in view(1, src))
+					if(tgt_dir == get_dir(src,F) && F.density && !F.welded)
+						visible_message("<span class='danger'>[src] pries open the firedoor!</span>")
+						F.open()
+
+	else
+		mylocation = m2
+		chasecycles = 0
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/ClearObstacle(turf/target_turf)
+	// ***** This does not allow spiders to smash windoors (e.g: UO71 bar windoor)
```

Comment:
`does not allow spiders to smash windoors`

`var/list/valid_obstacles = list(/obj/structure/window, /obj/structure/closet, /obj/structure/table, /obj/structure/grille, /obj/structure/rack, /obj/machinery/door/window)`
`/obj/machinery/door/window`
🤔